### DISCUSSION TITLE
OSError: [Errno 18] Invalid cross-device link

### DISCUSSION THREAD

**Discussion by ZhuZhuoHSPH** [liulab-dfci/lisa2#2] [Q&A]
Hello~

Thank you for this awesome software. I'm trying to run lisa for some mouse genes and got the following error:
`OSError: [Errno 18] Invalid cross-device link: '/tmp/tmpbipv10oa' -> '/home/eid/.conda/envs/conda-env-python3/lib/python3.7/site-packages/lisa/data/mm10/mm10_basic_2.1_RP_map.npz'`
and 
`lisa.lisa_core.DownloadRequiredError: Data is malformed or incomplete, run "lisa download [species]" to redownload dataset`.

I did `lisa download mm10` and still got the same errors. However, I had none of the problems when testing using `hg38`. I wonder if you could help me to figure out what the problem could be. Thank you very much!

Best,
Zhu

**Comment by AllenWLynch** [comment]
I am deeply sorry for my late response. I have found the problem in the backend database for mm10 and corrected it, and I have tested the data download myself and things seem to be back on track. Please let me know if you further feedback. If you would like to try the mm10 test again, I am pushing an update next week that will make the downloading much faster and adds functionality for providing your own regions-of-interest for the test!

**Comment by ZhuZhuoHSPH** [comment]
Thanks Allen, I haven't used it again, but I'll report back once I do. Thanks!

**Reply by liulab-dfci** [reply]
I am releasing an update of LISA today or tomorrow which will include a new motif database based on JASPAR, a new Gene-TF targeting matrix output (https://github.com/liulab-dfci/lisa2/blob/master/docs/DataAnalysisGuide.md#Which-Genes-do-my-Factors-Target) and support for supplying your own ATAC/DNase peaks for factor enrichment. When you try your analysis again, I highly recommend upgrading to 2.2.1! Thanks again

**Comment by ChelseaCHENX** [comment]
I debugged via this [post](https://stackoverflow.com/questions/42392600/oserror-errno-18-invalid-cross-device-link), for `lisa/core/data_interface.py`, line 70, `in fetch_from_cistrome
    os.rename(filename, cls.get_dataset_path(species, window_size))`, modify `os.rename` to `shutil.move` (and add `import shutil` to this file)

**Comment by AllenWLynch** [comment]
Thank you for pointing me to this issue. I was not aware that os.rename had this limitation. I will include this change in an update soon.
________________________________
